src/frequenz/client/reporting/_client.py

```diff
@@ -0,0 +1,289 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Client for requests to the Reporting API."""
+
+from collections import namedtuple
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, AsyncIterator, Awaitable, Generator, Type, cast
+
+import grpc.aio as grpcaio
+
+# pylint: disable=no-name-in-module
+from frequenz.api.common.v1.metrics.metric_sample_pb2 import Metric as PBMetric
+from frequenz.api.common.v1.microgrid.microgrid_pb2 import (
+    MicrogridComponentIDs as PBMicrogridComponentIDs,
+)
+from frequenz.api.common.v1.pagination.pagination_params_pb2 import (
+    PaginationParams as PBPaginationParams,
+)
+from frequenz.api.reporting.v1.reporting_pb2 import (
+    ListMicrogridComponentsDataRequest as PBListMicrogridComponentsDataRequest,
+)
+from frequenz.api.reporting.v1.reporting_pb2 import (
+    ListMicrogridComponentsDataResponse as PBListMicrogridComponentsDataResponse,
+)
+from frequenz.api.reporting.v1.reporting_pb2 import TimeFilter as PBTimeFilter
+from frequenz.api.reporting.v1.reporting_pb2_grpc import ReportingStub
+from frequenz.client.common.metric import Metric
+from google.protobuf.timestamp_pb2 import Timestamp as PBTimestamp
+
+# pylint: enable=no-name-in-module
+
+Sample = namedtuple("Sample", ["timestamp", "value"])
+"""Type for a sample of a time series."""
+
+MetricSample = namedtuple(
+    "MetricSample", ["timestamp", "microgrid_id", "component_id", "metric", "value"]
+)
+"""Type for a sample of a time series incl. metric type, microgrid and component ID"""
```

Comment:
In theory, we could just use `MetricSample` everywhere, instead of using `Sample`. The higher level layers could create bespokse structs based on what they see fit. That would keep this library simpler, and in the future, we could just extend `MetricSample` to hold states, bounds, and aggregated metrics (instead of having to do that for `Sample` as well).

Any thoughts?

Reply:
Yea, another way could be a `MetricSample` type from named tuple with all fields but timestamp and value optional. Depending on context the relevant fields would be set.

I would approach that in a next iteration where also the case of multiple metrics is implemented. As this is in a very early stage still, I think we can merge this for now.